### PR TITLE
Update OpenTelemetry Dependencies to latest

### PR DESCRIPTION
**Pull request by srv-gh-o11y-gdi**
Update OpenTelemetry Dependencies to latest

### CODE REVIEW

**Comment by samiura** [comment]
🤞🏽

**Review by pjanotti — APPROVED** [review]

**Review by samiura — APPROVED** [review]

File: pkg/processor/timestampprocessor/go.mod [hunk 1]
```diff
@@ -1,14 +1,16 @@
 module github.com/signalfx/splunk-otel-collector/pkg/processor/timestampprocessor
 
-go 1.21
+go 1.21.0
+
+toolchain go1.21.10
```
Comment (crobert-1) [inline pkg/processor/timestampprocessor/go.mod:5]: Same question, should we force a minimum version `go` toolchain here?

File: internal/tools/go.mod [hunk 2]
```diff
@@ -1,6 +1,8 @@
 module github.com/signalfx/splunk-otel-collector/internal/tools
 
-go 1.21
+go 1.21.0
+
+toolchain go1.21.10
```
Comment (crobert-1) [inline internal/tools/go.mod:5]: Is the `toolchain` directive required here?

Comment (pjanotti) [reply internal/tools/go.mod:5]: It was added by the tool itself, I didn't try remove it.

**Pull request merged**